packages/server/src/migrations/0.1.21-client-tokens.ts

```diff
@@ -0,0 +1,58 @@
+import { Migration } from '@botpress/messaging-engine'
+import { v4 as uuidv4 } from 'uuid'
+
+export class ClientTokensMigration extends Migration {
+  meta = {
+    name: ClientTokensMigration.name,
+    description: 'Transfers client tokens from msg_clients to a new table msg_client_tokens',
+    version: '0.1.21'
+  }
+
+  async valid() {
+    return this.trx.schema.hasTable('msg_clients')
+  }
+
+  async applied() {
+    return !(await this.trx.schema.hasColumn('msg_clients', 'token'))
+  }
+
+  async up() {
+    await this.trx.schema.createTable('msg_client_tokens', (table) => {
+      table.uuid('id').primary()
+      table.uuid('clientId').references('id').inTable('msg_clients')
+      table.string('token').notNullable()
+      table.timestamp('expiry').nullable()
+    })
```

Comment:
I added the creation of the table in the migration here. This actually needs to happen in the migration, otherwise migrations might be unstable if we take the latest format of that table, since this format could change in the future. I also added the deletion of the table in the down migration.

Reply:
IMHO, if decide to go against our standards here, we expose ourselves to future problems and inconsistencies across the repo. I still don't understand why you choose to put the table creation here considering that the "problem" in question is valid for all other tables too? Why only in the particular case? 

What do you mean by `migrations might be unstable if we take the latest format of that table`? If this format is to change in the future, it will be done by another migration that would implement an up and down function.

Reply:
We don't create the table at two places, we create the table in the migrations for down/up compatibility, and we have code in the server that can create the database for the current version of the server. The migrations can't depend on the code that creates the table in the server since that code can change. So now any creation or modification of table is going to imply a migration, and the only use for the table creation code in the server is to create a new database. See #299